src/schema.rs

```diff
@@ -197,4 +213,92 @@ mod schema_tests {
         let types: Vec<TypeRef> = schema.unwrap().get_types().collect();
         assert_eq!(types.len(), total_types);
     }
+
+    #[rstest(
+        valid_values, invalid_values, schema,
+        case::type_constraint(
+            load(r#"
+                5
+                0
+                -2
+            "#),
+            load(r#"
+                false
+                "hello"
+                5.4
+            "#),
+            load_schema_from_text(r#" // For a schema with named type as below: 
+                type:: { name: my_int, type: int }
+            "#),
+        ),
+        case::not_constraint(
+            load(r#"
+                true
+                "hello"
+                5.4
+                6e10
+            "#),
+            load(r#"
+                5
+                0
+                -1
+            "#),
+            load_schema_from_text(r#" // For a schema with not constraint as below: 
+                type:: { name: not_type, not: { type: int } }
+            "#),
+        ),
+        case::one_of_constraint(
+            load(r#"
+                5
+                -5
+                5.4
+                -5.4
+            "#),
+            load(r#"
+                false
+                "hello"
+                hey
+            "#),
+            load_schema_from_text(r#" // For a schema with one_of constraint as below: 
+                type:: { name: one_of_type, one_of: [int, decimal] }
+            "#),
+        ),
+        // TODO: add a test case for all_od constraint
+        case::any_of_constraint(
+            load(r#"
+                5
+                5.4
+                true
+            "#),
+            load(r#"
+                "hello"
+                hey
+                6e10
+            "#),
+            load_schema_from_text(r#" // For a schema with any_of constraint as below: 
+                type:: { name: any_of_type, any_of: [int, decimal, bool] }
+            "#),
+        ),
+    )]
+    fn type_validation(
+        valid_values: Vec<OwnedElement>,
+        invalid_values: Vec<OwnedElement>,
+        schema: Rc<Schema>,
+    ) {
+        let types: Vec<TypeRef> = schema.get_types().collect();
```

Comment:
Making a `Vec` and then referring to `types[0]` is a little less clear to me than:
```rust
// There is only a single type in each schema defined above.
let type_ref: TypeRef = schema.get_types().first().expect("Loaded schema was empty.");
```